From b9f5578e710a28880a02ae607c287ff53bd28ee2 Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <tlanger@maxlinear.com>
Date: Mon, 19 Apr 2021 19:19:45 +0200
Subject: [PATCH] PONRTSYS-8555: More cleanup in dp_mib_counters.c

Make functions static, use "int" instead of "s32" to return errors.
and fix the handling of wraparound (off-by-one problem)
---
 .../net/datapath/mib_counters/dp_mib_counters.c    | 25 ++++++++++++----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/drivers/net/datapath/mib_counters/dp_mib_counters.c b/drivers/net/datapath/mib_counters/dp_mib_counters.c
index 1d7399528a3f..5ada2afceedb 100644
--- a/drivers/net/datapath/mib_counters/dp_mib_counters.c
+++ b/drivers/net/datapath/mib_counters/dp_mib_counters.c
@@ -174,7 +174,7 @@ static u8 shortcut_mode;
 /* Wraparound counter handle */
 static u64 wraparound(u64 curr, u64 last, u32 size)
 {
-	#define WRAPAROUND_MAX_32 0xFFFFFFFF
+	#define WRAPAROUND_MAX_32 0x100000000
 	/* for 8 bytes(64bit mib),no need to do wraparound */
 	if ((size > 4) || (curr >= last))
 		return curr - last;
@@ -212,7 +212,7 @@ static void update_dev_counters(struct mib_counters *cnt, u8 initial_update)
 	u32 tx_good_pkts_ctp_bypass = 0;
 	u32 tx_cbm_pkts = 0;
 	s64 qos_drop_pkts = 0;
-	s32 ret = 0;
+	int ret = 0;
 	u8 i = 0;
 	u16 port_id = 0;
 	GSW_portType_t port_type = GSW_CTP_PORT;
@@ -455,7 +455,8 @@ static void update_all_devs_counters(void)
 }
 
 /* Adds new net device to counter update list */
-s32 add_dev_to_list(struct net_device *dev, s32 port_id, s32 sub_if, s32 inst)
+static int add_dev_to_list(struct net_device *dev,
+			   s32 port_id, s32 sub_if, s32 inst)
 {
 	struct mib_counters *dev_cnt;
 	struct pmac_port_info *port_info;
@@ -519,7 +520,8 @@ s32 add_dev_to_list(struct net_device *dev, s32 port_id, s32 sub_if, s32 inst)
 }
 
 /* Looks for matched list item and remove it */
-s32 rm_dev_from_list(struct net_device *dev, s32 port_id, s32 sub_if, s32 inst)
+static int rm_dev_from_list(struct net_device *dev,
+			    s32 port_id, s32 sub_if, s32 inst)
 {
 	struct pmac_port_info *port_info;
 	struct mib_counters *mib_cnt;
@@ -589,8 +591,9 @@ s32 rm_dev_from_list(struct net_device *dev, s32 port_id, s32 sub_if, s32 inst)
 }
 
 /* Function needed for ethtool to find on list appropriate wide counters set */
-struct mib_counters *find_registered_dev(s32 port_id, s32 sub_if_id_grp,
-					 s32 inst, struct net_device *dev)
+static struct mib_counters *find_registered_dev(s32 port_id, s32 sub_if_id_grp,
+						s32 inst,
+						struct net_device *dev)
 {
 	struct mib_counters *mib_cnt;
 	struct list_head *p;
@@ -618,7 +621,7 @@ struct mib_counters *find_registered_dev(s32 port_id, s32 sub_if_id_grp,
 }
 
 /* Callback function to register new net device (subif) reported by DP API */
-s32 subif_register_cb(struct dp_event_info *info)
+static s32 subif_register_cb(struct dp_event_info *info)
 {
 	DP_DEBUG(DP_DBG_FLAG_MIB,
 		 "MIB CNT - EVENT - SUBIF REGISTER CB called\n - event info type = %d\n",
@@ -694,8 +697,8 @@ static int update_lct_cnt(struct mib_counters_u64 *counters,
 
 /************************* ETHTOOL SUPPORT ************************/
 
-void mib_cnt_get_ethtool_stats_31(struct net_device *dev,
-				  struct ethtool_stats *stats, u64 *data)
+static void mib_cnt_get_ethtool_stats(struct net_device *dev,
+				      struct ethtool_stats *stats, u64 *data)
 {
 	struct pmac_port_info *port_info;
 	struct mib_counters *mib_cnt;
@@ -807,7 +810,7 @@ static void dp_dev_get_stat_strings(struct net_device *dev, u8 *data)
 
 /********************* workqueue handler ***************************/
 
-void cnt_upd_timer_handler(struct work_struct *work)
+static void cnt_upd_timer_handler(struct work_struct *work)
 {
 	/* Add delayed work object again for next run, the old object is not
 	 * changed any more.
@@ -859,7 +862,7 @@ static int __init mib_counters_mod_init(void)
 	shortcut_mode = ((gsw_mode & BIT(0)) == GSW_SHORTCUT_MODE);
 
 	/* Register ethtool stats cb function */
-	dp_set_ethtool_stats_fn(0, mib_cnt_get_ethtool_stats_31);
+	dp_set_ethtool_stats_fn(0, mib_cnt_get_ethtool_stats);
 	dp_set_ethtool_stats_strings_cnt_fn(0, dp_dev_get_stat_strings_count);
 	dp_set_ethtool_stats_strings_fn(0, dp_dev_get_stat_strings);
 
